Add support for local folders

inequation opened the pull request:
Hello!

Thanks for your extension, it does exactly what I expect from a P4 IDE integration. Here's a little contribution that allows it to work with "solutionless" folders, i.e. the "Open local folder" flow from the splash screen. This works because `IVsSolution` returns the folder path as solution name, as if it was the solution. To account for the fact that in this flow we have no way to store user preferences (like we do in the solution case), I've implemented detection of [`P4CONFIG` usage](https://www.perforce.com/manuals/v23.1/cmdref/Content/CmdRef/P4CONFIG.html), to make the automatic mode a more robust default.

One could argue that the `P4CONFIG` detection should run also in the solution case, as the subsequent user preference load from the solution should be able to overwrite the detected `P4CONFIG` defaults, but I don't need that in my case, so I'll leave that as a future opportunity.

inequation committed:
Upgrade target framework from 4.5 to 4.7.2

4.5 now shows an "unsupported" warning in VS.

inequation commented:
Oops, I've only noticed now that I've changed a ton of whitespace by inadvertently converting space indentation to tabs. 😬 If that's an issue, I can revert it. The diff will get much clearer if you click the cog icon and tick "Hide whitespace".

botman99 commented:
Yes, these look fine.  Please undo the whitespace changes and I can pull it in.

botman99 commented:
I think automatically checking for the p4config file even with a Solution file should be fine.  Also, I think the settings could be saved to a text file in the .vs folder after you've loaded a folder in VS.  I may work on adding both of those before publishing to the Microsoft Visual Studio marketplace.

inequation committed:
Implement IVsSolutionEvents7 to handle folder events as well as solutions

Allows using the extensions with "Open local folder" functionality. This works because IVsSolution returns the folder path as solution name. Also check for the existence of the P4CONFIG file in the folder and start in automatic mode if it's present.

inequation commented:
There, replaced the top commit with one with whitespace-only changes removed!

inequation commented:
Oh dear, I snuck a terrible bug in there. Hold on, fixing.

inequation committed:
Fix incorrect call to OnAfterOpenSolutionOrFolder instead of OnAfterCloseSolutionOrFolder

inequation committed:
Handle `P4CONFIG` also in the classic VS solution path

Also restructure `SccProvider.ReadUserOptions` to respect the user disabling the P4 connection in preferences.

inequation commented:
There, fixed a horrible bug and moved P4CONFIG handling to where it benefits both paths.
